vault/resource_pki_backend_role.go

```diff
@@ -0,0 +1,380 @@
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if clients can request certificates for localhost as one of the requested common names.",
+			},
+			"allowed_domains": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Policies to be set on tokens issued using this role.",
+			},
+			"allow_bare_domains": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if clients can request certificates matching the value of the actual domains themselves.",
+			},
+			"allow_subdomains": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if clients can request certificates with CNs that are subdomains of the CNs allowed by the other role options.",
+			},
+			"allow_glob_domains": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Allows names specified in allowed_domains to contain glob patterns.",
+			},
+			"allow_any_name": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if clients can request any CN.",
+			},
+			"enforce_hostnames": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if only valid host names are allowed for CNs, DNS SANs, and the host part of email addresses.",
+			},
+			"allow_ip_sans": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if clients can request IP Subject Alternative Names.",
+			},
+			"server_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if certificates are flagged for server use.",
+			},
+			"client_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if certificates are flagged for client use.",
+			},
+			"code_signing_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates are flagged for code signing use.",
+			},
+			"email_protection_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates are flagged for email protection use.",
+			},
+			"key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "rsa",
+				Description: "Specifies the type of key to generate for generated private keys. Currently, rsa and ec are supported.",
+			},
+			"key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     2048,
+				Description: "Specifies the number of bits to use for the generated keys.",
+			},
+			"key_usage": {
+				Type:     schema.TypeList,
```

Comment:
Is order relevant to the values here? Are the values going to be useful as interpolated values for other resources?

Reply:
Order isn't relevant here, no. Changed to a `schema.TypeSet`.

Yes, I think interpolation is useful for these, though I don't have a particular use case in mind.